dahdi-linux: forward-port to 6.18 kernel APIs, fix build flags, disable OSLEC

graysky2 opened the pull request:
Maintainer: no one
Compile tested: (put here arch, model, OpenWrt version) SNAPSHOT
Run tested: (put here arch, model, OpenWrt version, tests done) not run tested

Description:

This patchset updates DAHDI 3.4.0 to build cleanly on Linux 6.18 and modern OpenWrt toolchains. Changes include:

* Add kernel API compatibility shims:
  - Provide hrtimer_init() wrapper using hrtimer_setup()
  - Restore from_timer() helper
  - Map del_timer*() to timer_delete*() on >= 6.15, aligning with upstream PR #93[1].

* Replace deprecated EXTRA_CFLAGS with ccflags-y across all Kbuilds to match upstream kernel changes and resolve build failures. Relating to oct612x include paths using ccflags-y, aligning with upstream PR #76[2]
.
* Disable OSLEC echo canceller:
  - OpenWrt provides only oslec.h without echo.c implementation
  - Prevents undefined oslec_* symbol failures during modpost
  - Marks dahdi-echocan-oslec package as unavailable on >= 6.18

* Minor Makefile adjustments for OpenWrt packaging consistency.

These changes collectively restore a complete, warning-free build of DAHDI 3.4.0 on Linux 6.18 while preserving compatibility with existing drivers and OpenWrt module packaging.

1. https://github.com/asterisk/dahdi-linux/issues/93
2. https://github.com/asterisk/dahdi-linux/issues/76

Commit:
dahdi-linux: forward-port to 6.18 kernel APIs, fix build flags

This patchset updates DAHDI 3.4.0 to build cleanly on Linux 6.18 and
modern OpenWrt toolchains. Changes include:

* Add kernel API compatibility shims:
  - Provide hrtimer_init() wrapper using hrtimer_setup()
  - Restore from_timer() helper
  - Map del_timer*() to timer_delete*() on >= 6.15, aligning with
    upstream PR #93[1].

* Replace deprecated EXTRA_CFLAGS with ccflags-y across all Kbuilds
  to match upstream kernel changes and resolve build failures. Relating
  to oct612x include paths using ccflags-y, aligning with upstream
  PR #76.[2]

* Use out-of-tree OSLEC for Linux 6.18

* Minor Makefile adjustments for OpenWrt packaging consistency.

These changes collectively restore a complete, warning-free build of
DAHDI 3.4.0 on Linux 6.18 while preserving compatibility with existing
drivers and OpenWrt module packaging.

1. https://github.com/asterisk/dahdi-linux/issues/93
2. https://github.com/asterisk/dahdi-linux/issues/76

Signed-off-by: John Audia <therealgraysky@proton.me>
Signed-off-by: Daniel Golle <daniel@makrotopia.org>

dangowrt commented:
OSLEC used to work fine with older kernels, it is built as module `kmod-echo`. It would be nice if don't have to drop it, I'm actively using it with the HFC-S BRI driver (HFC-S is a low-cost BRI chip without any DSP functionality, without OSLEC there is no echo cancelation at all, which is the more annoying the more latency there is in a call)

cotequeiroz commented on libs/dahdi-linux/patches/211-Kbuild-Use-ccflag-y.patch:
```diff
@@ -0,0 +1,196 @@
+ obj-$(DAHDI_BUILD_ALL)$(CONFIG_DAHDI_OPVXA1200) += opvxa1200.o
+ 
+-EXTRA_CFLAGS += -I$(src)/.. -Wno-undef -Wno-error=unused-variable
++ccflags-y += -I$(src)/.. -Wno-undef
```
You're dropping  `-Wno-error=unused-variable`, but it is needed, at least for mips:
```
opvxa1200/base.c: In function 'wctdm_transmitprep':
opvxa1200/base.c:467:32: error: unused variable 'pos' [-Werror=unused-variable]
  467 |         int x, y, chan_offset, pos;
      |                                ^~~
opvxa1200/base.c:467:13: error: unused variable 'x' [-Werror=unused-variable]
  467 |         int x, y, chan_offset, pos;
      |             ^
opvxa1200/base.c: In function 'wctdm_receiveprep':
opvxa1200/base.c:556:16: error: unused variable 'y' [-Werror=unused-variable]
  556 |         int x, y, chan_offset;
      |                ^
cc1: all warnings being treated as errors
```

dangowrt committed:
oslec: new package

Out-of-tree build of the Open Source Line Echo Canceller, formerly
shipped as drivers/misc/echo in the kernel and removed upstream.
Provides oslec_* symbols for dahdi_echocan_oslec on kernels >= 6.18.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>
